src/frontend/Sider/Behandling/Totrinnskontroll/FatteVedtak.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+import { FormEvent, useState } from 'react';
+
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+import {
+    Alert,
+    BodyShort,
+    Button,
+    Checkbox,
+    CheckboxGroup,
+    Heading,
+    Radio,
+    RadioGroup,
+    Textarea,
+} from '@navikt/ds-react';
+
+import { ÅrsakUnderkjent, årsakUnderkjentTilTekst } from './typer';
+import { useApp } from '../../../context/AppContext';
+import { useBehandling } from '../../../context/BehandlingContext';
+import { RessursStatus } from '../../../typer/ressurs';
+
+const WrapperMedMargin = styled.div`
+    display: block;
+    margin: 0.5rem 0;
+`;
+
+const SubmitButtonWrapper = styled.div`
+    display: flex;
+    justify-content: center;
+    margin-top: 0.5rem;
+`;
+
+const TittelContainer = styled.div`
+    margin: 0.5rem 0;
+`;
+
+interface TotrinnskontrollForm {
+    godkjent: boolean;
+    begrunnelse?: string;
+    årsakerUnderkjent: ÅrsakUnderkjent[];
+}
+
+enum Totrinnsresultat {
+    IKKE_VALGT = 'IKKE_VALGT',
+    GODKJENT = 'GODKJENT',
+    UNDERKJENT = 'UNDERKJENT',
+}
+
+const FatteVedtak: React.FC<{
+    settVisGodkjentModal: (vis: boolean) => void;
+}> = ({ settVisGodkjentModal }) => {
+    const { request } = useApp();
+    const navigate = useNavigate();
+    const { behandling } = useBehandling();
+
+    const [godkjent, settGodkjent] = useState<Totrinnsresultat>(Totrinnsresultat.IKKE_VALGT);
+    const [årsakerUnderkjent, settÅrsakerUnderkjent] = useState<ÅrsakUnderkjent[]>([]);
+    const [begrunnelse, settBegrunnelse] = useState<string>();
+    const [feil, settFeil] = useState<string>();
+    const [laster, settLaster] = useState<boolean>(false);
+
+    const erUtfylt =
+        godkjent === Totrinnsresultat.GODKJENT ||
+        (godkjent === Totrinnsresultat.UNDERKJENT &&
+            (begrunnelse || '').length > 0 &&
```

Comment:
Dette er vel ekvivalent?
```suggestion
        (godkjent === Totrinnsresultat.UNDERKJENT &&
            begrunnelse &&
```